spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -653,8 +654,9 @@ Order<Product> order =
         Order.by(_Product.price.asc(),
                  _Product.id.asc());
 PageRequest pageRequest =
-        PageRequest.ofSize(10)
-                   .afterKey(priceMidpoint, 0L);
+        PageRequest.ofPage(5)
```

Comment:
The same is true here; maybe I did not get your point, or I am missing some details.

But once we are talking about the cursor, we should not define the page number.

Reply:
> But once we are talking about the cursor, we should not define the page number.

I disagree. If this thing always carries a page number, the page number should have a sensible value.

(We have not even defined a magical value like `-1` to mean "the page number is nonsense".)

Reply:
@gavinking Sorry, but I don't follow you.

My point is that the demo focuses on a cursor that won't use the page number. Thus, why do we need to put something like this:

```java
 PageRequest.ofSize(10).afterCursor(Cursor.forKey(priceMidpoint, 0L));
```

⚠️ I am not talking about putting a magical number, but I am going to the point and putting information that the cursor requires, such as the cursor and the size, skipping the page.

Reply:
When that `PageRequest` is ultimately used to construct a `CursoredPageRecord`, it matters whether it represents a "first" page. 

- If it _does_ then previous page navigation should result in an offset-based `PageRequest`. 
- It it _doesn't_, then previous page navigation should result in a cursor-based `PageRequest`.

Completely apart from that, `page()` is a public member of `PageRequest`, exposed to the application programmer, and shouldn't return an incorrect value.